tools/analyze_results.py

```diff
@@ -0,0 +1,242 @@
+
+        good_dir = osp.abspath(osp.join(self.show_dir, 'good'))
+        bad_dir = osp.abspath(osp.join(self.show_dir, 'bad'))
+        self._save_image_gts_results(good_mAPs, good_dir)
+        self._save_image_gts_results(bad_mAPs, bad_dir)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='MMDet eval image prediction result for each')
+    parser.add_argument('config', help='test config file path')
+    parser.add_argument(
+        'prediction_path', help='prediction path where test pkl result')
+    parser.add_argument(
+        'show_dir', help='directory where painted images will be saved')
+    parser.add_argument('--show', action='store_true', help='show results')
+    parser.add_argument(
+        '--wait-time',
+        type=float,
+        default=0,
+        help='the interval of show (s), 0 is block')
+    parser.add_argument(
+        'eval',
+        type=str,
+        help='evaluation metrics, which depends on the dataset, e.g., "bbox",'
+        ' "segm", "proposal" for COCO, and "mAP", "recall" for PASCAL VOC')
+    parser.add_argument(
+        '--topk',
+        default=20,
+        type=int,
+        help='Saved Number of the highest topk '
+        'and lowest topk after index sorting')
+    parser.add_argument(
+        '--cfg-options',
+        nargs='+',
+        action=DictAction,
+        help='override some settings in the used config, the key-value pair '
+        'in xxx=yyy format will be merged into config file. If the value to '
+        'be overwritten is a list, it should be like key="[a,b]" or key=a,b '
+        'It also allows nested list/tuple values, e.g. key="[(a,b),(c,d)]" '
+        'Note that the quotation marks are necessary and that no white space '
+        'is allowed.')
+    parser.add_argument(
+        '--eval-options',
+        nargs='+',
+        action=DictAction,
+        help='custom options for evaluation, the key-value pair in xxx=yyy '
+        'format will be kwargs for dataset.evaluate() function')
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+
+    mmcv.check_file_exist(args.prediction_path)
+
+    cfg = Config.fromfile(args.config)
+    if args.cfg_options is not None:
+        cfg.merge_from_dict(args.cfg_options)
+    cfg.data.test.test_mode = True
+    # import modules from string list.
+    if cfg.get('custom_imports', None):
+        from mmcv.utils import import_modules_from_strings
+        import_modules_from_strings(**cfg['custom_imports'])
+
+    cfg.data.test.pop('samples_per_gpu', 0)
+    cfg.data.test.pipeline = retrieve_loading_pipeline(cfg.data.train.pipeline)
+    dataset = build_dataset(cfg.data.test)
+    outputs = mmcv.load(args.prediction_path)
+    result_visualizer = ResultVisualizer(
+        dataset,
+        outputs,
+        topk=args.topk,
+        show=args.show,
+        wait_time=args.wait_time,
+        show_dir=args.show_dir)
+    result_visualizer.evaluate_and_show()
```

Comment:
It seems that the visualizer is general and can be used many times after creation. If so, we may not make dataset and outputs as an attribute to initialize the ResultVisualizer, but make these two arguments of `evaluate_and_show()`.